### DISCUSSION TITLE
MPFS-Video-Kit Reference Design customisation

### DISCUSSION THREAD

**Discussion by TobiasPaulikat** [polarfire-soc/polarfire-soc-documentation#456] [How to]
I'm currently trying to implement the H.264 Encoding Reference Design from the Video-Kit to an Disco-Kit. 
When running the Webserver with the "Start" button, I get the following error message: 
[29.347412] mchp-vcpp 60000000.video-pipeline: dma alloc of size 2060288 failed

Since 2060288 should be 2 Gb and the Disco-Kit only have 1 Gb RAM I want to test if I can simply change the DMA size. 
Can you tell me where I need to set the Value ?

Regards

**Comment by sc-mchp** [comment]
Hi,

Even if you got this error, The demo should work.

still if you want to remove this error in the log.
you can remove the below line the dtso file, dma alloc failed error will not show.
https://github.com/linux4microchip/dt-overlay-mchp/blob/master/mpfs_video/mpfs_modular_h264_pipeline.dtso#L64

Thanks,
Shravan

**Comment by TobiasPaulikat** [comment]
Hi Shravan, 

Thank you!

I think the Problem is the Kamera then. 
I'm using an IMX219 Sensor with the Raspberry Pi V2 Kamera and changed the Adress from 0x1a to 0x10 in the Devicetree and the ffmpeg.sh 
Still I dont get any Video or Frame out of it. 

If I test it with the Bayer Pipeline (only MIPI to VDMA) I get the Error Message : Time out

Do you have any Idea ?

Regards,
Tobi

**Reply by sc-mchp** [reply]
Hi Tobi,

we have not tested any other camera with mpfs-video-kit. 
the pipeline expecting 1080p resolution with 4 lane. 
If you have changed the camera need to change the device tree entry for your camera and Libero design if required. 

Thanks,
Shravan

**Reply by TobiasPaulikat** [reply]
Hi Shravan, 

I'm using the IMX219 Sensor with 2 Lane MIPI in 1080p. I changed the device tree and the libero design. 
I'm getting strange Clockfrequency. There is an option for continous clock mode in an register wich is not listed in the driver (linux4microchip/linux/media/i2c/imx219.c) is there any chance to set this register, when the device is booted or can I set this specific register in the device driver ?

Thank you

**Reply by sc-mchp** [reply]
I have not worked with imx219.c driver any time, i do not know about "continous clock mode"
it is better to edit the driver, if you want any changes
because the driver will write all register values every time you start capture command from application.

**Reply by TobiasPaulikat** [reply]
Okay thank you! 
For your Help, I'll try to find out how I change it.

**Comment by Alex-e-lab** [comment]
@TobiasPaulikat have you any success running imx219 on Discovery kit?
